scripts/Set-AzureAdApplication/v2/Set-AadApplication.psm1

```diff
@@ -150,6 +151,31 @@ function Set-AadApplication {
             }
         }
 
+        # Check for existing secrets and remove them so they can be re-created
+        Write-Verbose "Checking for existing secrets"
+        $appKeySecrets = Get-AzureADApplicationPasswordCredential -ObjectId $application.ObjectId
+
+        if($appKeySecrets)  {
+            foreach($existingSecret in $appKeySecrets) {
+                foreach($secret in $Secrets) {
+                    if([System.Text.Encoding]::ASCII.GetString($existingSecret.CustomKeyIdentifier) -eq $secret.Description) {
+                        Write-Verbose "Removing existing key with description: $secret.Description"
+                        Remove-AzureADApplicationPasswordCredential  -ObjectId $application.ObjectId -KeyId $existingSecret.KeyId
+                    }
+                }
+            }
+        }
+
+        # Create new secrets
+        foreach($secret in $Secrets) {
+            $endDate = [datetime]::ParseExact($secret.EndDate,'dd/MM/yyyy',[Globalization.CultureInfo]::InvariantCulture)
+            Write-Verbose "Creating new key with description: $secret.Description and end date $secret.EndDate"
```

Comment:
Will $secret.Description print the value in the log? Shouldn't it be: $($secret.Description).
Same for Enddate

Reply:
Seems to be working just fine.
![image](https://user-images.githubusercontent.com/5137829/56942886-46f20300-6b1d-11e9-9cb9-b6d31d4c6d21.png)